src/main/java/Parser.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Represents a parser that helps to make sense of the user commands.
+ */
+public class Parser {
+    /**
+     * Reads the user input and returns the corresponding command according to the command
+     * given by the user.
+     *
+     * @param input The user input into the program.
+     * @return The corresponding command for the user input.
+     * @throws DukeException
+     */
+    public static Command parse(String input) throws DukeException {
```

Comment:
This method is doing a little too much work, perhaps can consider using a switch statement and defer the input parsing logic to separate functions